src/pyeval.jl

```diff
@@ -202,32 +198,37 @@ pasted into the Python code.   This allows you to evaluate code
 where the code itself is generated by a Julia expression.
 """
 macro py_str(code, options...)
+    m = :(pynamespace($__module__))
+    fname = String(__source__.file)
+    esc(:(PyCall.@_py_str($m, $m, $fname, $code, $(options...))))
```

Comment:
Why is this call to `esc` necessary, since there are `esc` calls within `@_py_str`? 

Indeed, the call to `esc` here seems like it might cause problems, because it prevents the names `pylocals`, `pyglobals`, and `ret` from being hygienized.

Reply:
No, `pylocals` etc... get hygenized by `@_py_str`, the `esc` here makes it so the interpolated variables don't also get hygenized, e.g. with the current code: 

```julia
julia> @macroexpand py"1+$i"
quote
    #= /home/marius/src/pyjulia/dev/PyCall/src/pyeval.jl:229 =#
    (var"#18#pyglobals", var"#19#pylocals") = (PyCall.pynamespace(Main), PyCall.pynamespace(Main))
    #= /home/marius/src/pyjulia/dev/PyCall/src/pyeval.jl:230 =#
    begin
        var"#19#pylocals"["__julia_localvar_2_1"] = PyCall.PyObject(i)
    end
    #= /home/marius/src/pyjulia/dev/PyCall/src/pyeval.jl:231 =#
    var"#20#ret" = (PyAny)(PyCall.pyeval_((string)("1+__julia_localvar_2_1"), var"#18#pyglobals", var"#19#pylocals", 258, "none"))
    #= /home/marius/src/pyjulia/dev/PyCall/src/pyeval.jl:232 =#
    begin
        PyCall.delete!(var"#19#pylocals", "__julia_localvar_2_1")
    end
    #= /home/marius/src/pyjulia/dev/PyCall/src/pyeval.jl:233 =#
    var"#20#ret"
end
```

Reply:
Maybe I should've asked this before but why [is `insert_pyevals` run at run-time](https://github.com/JuliaPy/pyjulia/pull/381/files#diff-edf2cf9175b9b67d834d1b52cd15a1dfL83)?  It looks like this doesn't depend on the run-time value of `globals` and `locals` (or maybe I'm missing something)?  If we expand everything at macro-expansion time, perhaps `esc` of `@py_str` here and `@prepare_for_pyjulia_call` there can be removed? Then maybe `_py_str` doesn't have to be a macro?